inspire_schemas/builders/references.py

```diff
@@ -218,17 +218,17 @@ def add_refextract_authors_str(self, authors_str):
 
     def add_author(self, full_name, role=None):
         self._ensure_reference_field('authors', [])
-
-        if role is not None:
-            inspire_role = 'editor' if role == 'ed.' else role
-            self.obj['reference']['authors'].append({
-                'full_name': normalize_name(full_name),
-                'inspire_role': inspire_role,
-            })
-        else:
-            self.obj['reference']['authors'].append({
-                'full_name': normalize_name(full_name),
-            })
+        if normalize_name(full_name):
```

Comment:
```suggestion
        normalized_name = normalize_name(full_name)
        if not normalized_name:
            return
```
so you don't need to indent more and can reuse the variable. Also move it before the `self._ensure_reference_field` as it will create an empty author list if we don't end up adding any authors.

Reply:
:+1: